client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
-jest.mock( 'calypso/blocks/dismissible-card', () => {
-	return function DismissibleCard() {
-		return null;
-	};
-} );
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import preferencesReducer from 'calypso/state/preferences/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import { Banner } from '../index';
 
-jest.mock( 'calypso/lib/analytics/track-component-view', () => {
-	return function TrackComponentView() {
-		return null;
+jest.mock( 'calypso/state/analytics/actions', () => {
+	return {
+		recordTracksEvent: jest.fn( () => ( {
+			type: 'ANALYTICS_EVENT_RECORD',
+		} ) ),
 	};
 } );
 
-import { Card, Button } from '@automattic/components';
-import { shallow } from 'enzyme';
-import PlanPrice from 'calypso/my-sites/plan-price/';
-import { Banner } from '../index';
+function renderWithRedux( ui ) {
+	return renderWithProvider( ui, {
+		reducers: {
+			preferences: preferencesReducer,
+		},
+		initialState: { preferences: { remoteValues: {} } },
+	} );
+}
 
 const props = {
 	callToAction: null,
 	title: 'banner title',
 };
 
 describe( 'Banner basic tests', () => {
-	test( 'should not blow up and have proper CSS class', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner' ) ).toHaveLength( 1 );
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'should output Banner component with proper CSS class', () => {
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.firstChild ).toHaveClass( 'banner' );
 	} );
 
 	test( 'should render Card if dismissPreferenceName is null', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ null } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } dismissPreferenceName={ null } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveClass( 'is-dismissible' );
 	} );
 
 	test( 'should render DismissibleCard if dismissPreferenceName is defined', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 0 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 1 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
```

Comment:
Nit: generally, when providing strings as props, it's more idiomatic to provide it just like we provide an attribute to an HTML element. That way we also avoid the extra evaluation that React performs - because everything inside the `{}` is a JS expression that is further evaluated.

```suggestion
		renderWithRedux( <Banner { ...props } dismissPreferenceName="banner-test" /> );
```